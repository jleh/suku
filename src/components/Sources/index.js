import React from 'react';
import { Translate } from 'react-localize-redux';

import styles from './sources.css';

const getSourceText = source => (
  <>
    <i>{source.author}</i> {source.title} <Translate id="page" /> {source.page}
  </>
);

const renderSourceLink = source => (
  <a href={source.url} target="_blank" rel="noopener noreferrer">
    {getSourceText(source)}
  </a>
);

const renderSource = source => (
  <div key={`${source.title}-${source.page}`}>
    {source.url ? renderSourceLink(source) : getSourceText(source)}
  </div>
);

export default ({ sources }) => sources && (
  <div className={styles.sources}>
    <h3><Translate id="sources" /></h3>
    {sources.map(renderSource)}
  </div>
);
