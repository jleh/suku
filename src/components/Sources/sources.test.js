import React from 'react';
import renderer from 'react-test-renderer';

import Sources from '.';

test('Renders source list', () => {
  const sources = [
    {
      title: 'Source title',
      page: '123',
      author: 'Author'
    }
  ];

  const tree = renderer
    .create(<Sources sources={sources} />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});

test('Renders source list with links', () => {
  const sources = [
    {
      title: 'Source title',
      page: '123',
      author: 'Author',
      url: 'http://www.example.com'
    }
  ];

  const tree = renderer
    .create(<Sources sources={sources} />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
