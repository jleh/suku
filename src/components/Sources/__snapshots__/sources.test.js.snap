// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders source list 1`] = `
<div
  className="sources"
>
  <h3>
    Missing translationId: sources for language: \${ languageCode }
  </h3>
  <div>
    <i>
      Author
    </i>
     
    Source title
     
    Missing translationId: page for language: \${ languageCode }
     
    123
  </div>
</div>
`;

exports[`Renders source list with links 1`] = `
<div
  className="sources"
>
  <h3>
    Missing translationId: sources for language: \${ languageCode }
  </h3>
  <div>
    <a
      href="http://www.example.com"
      rel="noopener noreferrer"
      target="_blank"
    >
      <i>
        Author
      </i>
       
      Source title
       
      Missing translationId: page for language: \${ languageCode }
       
      123
    </a>
  </div>
</div>
`;
